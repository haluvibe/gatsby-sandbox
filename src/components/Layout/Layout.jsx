import React, { Fragment, Component } from 'react';
import Section from '../Section/Section';
import Grid from '../Grid/Grid';
import Text from '../Text/Text';

const components = {
  section: Section,
  grid: Grid,
  text: Text,
};

export default ({component, ...others}) => {
  const ComponentType = components[component];
  return ComponentType 
    ? <ComponentType {...others} />
    : <div>{`${component} doesn't exist`}</div>
};