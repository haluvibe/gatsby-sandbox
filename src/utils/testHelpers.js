export const mockSettings = {
  id: 'id',
  className: 'a-class another-class',
  style: {
    backgroundColor: 'red',
    fontSize: 16,
  }
}

export const mockHtml = '<div>Hello test world</div>';