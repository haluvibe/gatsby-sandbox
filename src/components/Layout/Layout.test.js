import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { mockSettings } from '../../utils/testHelpers';

import Layout from './Layout';
const mockComponent = 'section';
const mockNonExistingComponent = 'hello world';

describe('<Layout />', () => {
  it('renders as expected', () => {
    const component = shallow(
        <Layout settings={mockSettings} component={mockComponent} />
      ),
      tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('handles missing component correctly', () => {
    const component = shallow(
        <Layout settings={mockSettings} component={mockNonExistingComponent} />
      ),
      tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
