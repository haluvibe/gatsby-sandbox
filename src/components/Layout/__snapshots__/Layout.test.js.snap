// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render as expected 1`] = `
<div>
  undefined doesn't exist
</div>
`;
